cwl/src/main/scala/cwl/internal/CwlEcmaScriptDecoder.scala

```diff
@@ -38,7 +38,8 @@ class CwlEcmaScriptDecoder {
       case long: java.lang.Long => WomLong(long).valid
       case double: java.lang.Double if double == double.doubleValue.floor && !double.isInfinite =>
         WomInteger(double.intValue).valid
-      case double: java.lang.Double => WomFloat(double).valid
+      case double: java.lang.Double => WomFloat(BigDecimal(double)).valid
+      case bigDecimal: java.math.BigDecimal => WomFloat(bigDecimal).valid
```

Comment:
Why `java.math.BigDecimal`, not `scala.math.BigDecimal`? 
I also think that it would be better to import this class and remove path declaration. Same applies to other classes here.

Reply:
@SergeySdv https://www.matheus.ro/2017/12/11/clean-code-boy-scout-rule/